packages/cloud_firestore/cloud_firestore/lib/src/field_value.dart

```diff
@@ -26,7 +26,7 @@ class FieldValue extends platform.FieldValuePlatform {
   /// will be overwritten with an array containing exactly the specified
   /// elements.
   static FieldValue arrayUnion(List<dynamic> elements) =>
-      FieldValue._(_factory.arrayUnion(elements));
+      FieldValue._(_factory.arrayUnion(_CodecUtility.valueEncode(elements)));
```

Comment:
We've had a bunch of those now, do we need to `.valueEncode` eeeeeverywhere that gets passed in to the platform?

Reply:
Yup, since we have different types between `cloud_firestore` and  `cloud_firestore_platform_interface`. The codec used to work with the method channel only accepts `cloud_firestore_platform_interface` which is the main reason we have to resort to encoding like that.

Reply:
Let me ask the question in a different way. Can we do this _CodecUtility.valueEncode everywhere that *might* be needed, before people cut a ticket with some other corner case, or have we plugged all the possible holes already? :P

Reply:
I keep thinking that we plugged all the hole until someone comes up with a different way of using Firestore than me 😅 . To be fair, the main focus for me was the integration tests that's why I keep adding to them whenever an issue pop up. 

It might be worth checking all the possible usage scenarios again and ensure that everything is encoded properly. I will start on that ASAP and get any fixes across if needed. (Hopefully nothing is needed though 😄 )

Reply:
Makes sense, I like the testing argument better than actually trying to plug everything at once! :P Keep it up!